prow/plugins/projectmanager/projectmanager_test.go

```diff
@@ -0,0 +1,678 @@
+					},
+				},
+				OrgProjects: map[string][]github.Project{},
+				ProjectColumnsMap: map[string][]github.ProjectColumn{
+					"testProject": {
+						{
+							Name: "testColumn2",
+							ID:   2,
+						},
+						{
+							Name: "testColumn",
+							ID:   1,
+						},
+					},
+				},
+				ColumnCardsMap: map[int][]github.ProjectCard{},
+				ColumnIDMap:    columnIDMap,
+			},
+			projectManager: plugins.ProjectManager{
+				OrgRepos: map[string]plugins.ManagedOrgRepo{
+					"testOrg/testRepo": {
+						Projects: map[string]plugins.ManagedProject{
+							"testProject": {
+								Columns: []plugins.ManagedColumn{
+									{
+										ID:     columnID,
+										State:  "open",
+										Org:    "otherOrg",
+										Labels: []string{"label1", "label2"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			pe: github.PullRequestEvent{
+				Action: github.PullRequestActionOpened,
+				Number: 1,
+				PullRequest: github.PullRequest{
+					ID:    2,
+					State: "open",
+				},
+				Repo: github.Repo{
+					Name: "someRepo",
+					Owner: github.User{
+						Login: "otherOrg",
+					},
+				},
+			},
+			expectedColumnCards: map[int][]github.ProjectCard{
+				1: {
+					{
+						ContentID:   2,
+						ContentType: "PullRequest",
+					},
+				},
+			},
+		},
+		{
+			name: "don't add pull request with incorrect labels",
+			gc: &fakegithub.FakeClient{
+				IssueLabelsAdded:   labels,
+				IssueLabelsRemoved: []string{},
+				RepoProjects: map[string][]github.Project{
+					"testOrg/testRepo": {
+						{
+							Name: "testProject",
+							ID:   1,
+						},
+					},
+				},
+				OrgProjects: map[string][]github.Project{},
+				ProjectColumnsMap: map[string][]github.ProjectColumn{
+					"testProject": {
+						{
+							Name: "testColumn2",
+							ID:   2,
+						},
+					},
+					"testProject2": {
+						{
+							Name: "testColumn2",
+							ID:   4,
+						},
+						{
+							Name: "testColumn",
+							ID:   3,
+						},
+					},
+				},
+				ColumnCardsMap: map[int][]github.ProjectCard{},
+			},
+			projectManager: plugins.ProjectManager{
+				OrgRepos: map[string]plugins.ManagedOrgRepo{
+					"testOrg/testRepo": {
+						Projects: map[string]plugins.ManagedProject{
+							"testProject": {
+								Columns: []plugins.ManagedColumn{
+									{
+										Name:   "testColumn",
+										State:  "open",
+										Org:    "otherOrg",
+										Labels: []string{"label1", "label2"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			pe: github.PullRequestEvent{
+				Action: github.PullRequestActionOpened,
+				Number: 1,
+				PullRequest: github.PullRequest{
+					ID:    2,
+					State: "open",
+				},
+				Repo: github.Repo{
+					Name: "someRepo",
+					Owner: github.User{
+						Login: "otherOrg",
+					},
+				},
+			},
+			expectedColumnCards: map[int][]github.ProjectCard{},
+			expectedError:       fmt.Errorf("could not find column %s in project %s", "testColumn", "testProject"),
```

Comment:
It is unclear to me why this testcase called "don't add pull request with incorrect labels" would result in an error. We don't want to emit errors on perfectly valid PRs that don't need to be touched by this plugin.

Reply:
Changed the test name, I changed code to not return error but ignore adding the PR to the project if that makes sense. It is possible to drop columns from projects after we configure and there is possibility that we cannot add the project card, we should ignore and log the details.